README.md

```diff
@@ -1,12 +1,21 @@
-# Realm Shim
+# Realms Shim : OBSOLETE, INSECURE, NOT RECOMMENDED FOR USE
+
 [![Build Status][circleci-svg]][circleci-url]
 [![Coverage Status][coveralls-svg]][coveralls-url]
 [![dependency status][deps-svg]][deps-url]
 [![dev dependency status][dev-deps-svg]][dev-deps-url]
 [![License][license-image]][license-url]
 
+This repository contains a shim implementation of the [Realm API Proposal](https://github.com/tc39/proposal-realms/#ecmascript-spec-proposal-for-realms-api). 
+
+The Realms proposal has been superceded by the [ShadowRealms Proposal](https://github.com/tc39/proposal-shadowrealm/). This shim does not implement ShadowRealms, but (as of this writing) the proposal is at Stage 3 of the TC39 standardization process, so common JS engines should have support soon.
+
+We do not recommend the use of this shim or the original Realms approach. The main danger is that intrinsics are not automatically frozen, so when two mutually-suspicious Realms are communicating, both must be extremely careful to prevent any object leakage, otherwise one Realm can pollute the globals or intrinsics of the other. This can be prevented with a "Membrane Pattern", but the shim doesn't provide one, and the details are tricky to get right.
```

Comment:
I'm wondering if we're taking the right approach in explaining the problem here. `Realm` is a valid API for generating a new set of intrinsics, aka a new sandbox. However it provides nothing to prevent the sandbox from leaking, which can lead to unexpected modifications when objects are shared between sandboxes.

There are really 2 way to make the sandbox safe: preventing objects from leaking in the first place (membrane), or making sure any leaked object cannot be mutated (hardened).

I'm wondering if we could simply say that while the `Realm` API allows the user to create a new set of independent built-ins (intrinsics), it provides no security feature without a strong isolation mechanism layered on top, which this shim does not provide.

Reply:
I'll rewrite the second paragraph to make that point, but I think I'll retain the conclusion, because we can't recommend using the shim *or* Realms in general, for security/isolation purposes, without doing so in conjunction with a second tool (a way to freeze intrinsics, or a membrane to protect the unfrozen ones), and the shim doesn't provide that second tool, and the second tool is non-trivial to construct.